### PR TITLE
Fixes #19977: Tags of a rule cannot be deleted

### PR DESCRIPTION
**Pull request by RaphaelGauthier**
https://issues.rudder.io/issues/19977

### CODE REVIEW

File: webapp/sources/rudder/rudder-web/src/main/elm/rules/sources/ViewTabContent.elm [hunk 1]
```diff
@@ -28,17 +28,23 @@ buildListCategories sep c =
...
     tagsList = List.map (\t ->
-      div [class "btn-group btn-group-xs"]
+      div [class "btn-group btn-group-xs delete-action"]
         [ button [class "btn btn-default tags-label", type_ "button"]
           [ i[class "fa fa-tag"][]
           , span[class "tag-key"][text t.key]
           , span[class "tag-separator"][text "="]
           , span[class "tag-value"][text t.value]
           ]
+        , (if editMode == True then
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-web/src/main/elm/rules/sources/ViewTabContent.elm:42]: please stop doing "boolean value" == True, just if editMode

File: webapp/sources/rudder/rudder-web/src/main/elm/rules/sources/ViewTabContent.elm [hunk 2]
```diff
@@ -115,10 +121,10 @@ tabContent model details =
                     [ input[ id "rule-tags-key", type_ "text", placeholder "key", class "form-control", onInput (\s -> UpdateRuleForm {details | ui = {ui | newTag = {newTag | key = s}}} ), value newTag.key][]
                     , span [ class "input-group-addon addon-json"][ text "=" ]
                     , input[ type_ "text", placeholder "value", class "form-control", onInput (\s -> UpdateRuleForm {details | ui = {ui | newTag = {newTag | value = s}}}), value newTag.value][]
-                    , span [ class "input-group-btn"][ button [ class "btn btn-success", type_ "button", onClick  (UpdateRuleForm {details | rule = {rule |  tags = newTag :: rule.tags }}) ][ span[class "fa fa-plus"][]] ]
+                    , span [ class "input-group-btn"][ button [ class "btn btn-success", type_ "button", onClick  (UpdateRuleForm {details | rule = {rule |  tags = newTag :: rule.tags }, ui = {ui | newTag = {newTag | key = "", value = "" }}}), disabled (String.isEmpty details.ui.newTag.key || String.isEmpty details.ui.newTag.value) ][ span[class "fa fa-plus"][]] ]
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-web/src/main/elm/rules/sources/ViewTabContent.elm:124]: you don't need to copy newTag here: 
```ui = {ui | newTag = {newTag | key = "", value = "" }} ```

you could 

```ui = {ui | newTag = Tag "" ""}```

(or the correct type, but don't need to make it so complicated)

**Comment by RaphaelGauthier** [comment]
Commit modified

**Review by VinceMacBuche — APPROVED** [review]

**Comment by RaphaelGauthier** [comment]
OK, merging this PR

**Pull request merged**